src/Microsoft.PowerShell.Commands.Utility/commands/utility/Sort-Object.cs

```diff
@@ -238,6 +239,13 @@ private int Heapify(List<OrderByPropertyEntry> dataToSort, OrderByPropertyCompar
         /// </summary>
         protected override void EndProcessing()
         {
+            // #17953: PSCustomObject contains only NoteProperties ignored by -Unique by default
+            // If first object is PSCustomObject, then default to "*" instead
+            if (Unique && Property?.Any() != true && InputObjects.FirstOrDefault()?.BaseObject is PSCustomObject)
+            {
+                Property = new[] { "*" };
+            }
```

Comment:
I doubt that this is correct for sorting, because unlike Select-Object we process all objects at once.

Reply:
So? That's why it's in `EndProcessing`, just before the `Property` is passed to the sorter. Without this change the tests fail.

Reply:
I mean that perhaps it is not correct to look only first element for sorting scenario. What if only first element is not PSCustomObject?

Reply:
This was the recommendation of the WG in the issue. I agree it may be a bit weird, since for a mix of types, their order will matter - something that should never happen with a sorting function. I guess we should go back to the issue to discuss it, I will make a comment asking WG for extra input.

Reply:
What is the status of this PR? 

I can add it to the agenda for the next cmdlet WG meeting. But not too sure what advice you are looking for! :-) I see Ilya's point. What should happen here?

Reply:
@doctordns I think we should go back to the drawing board and think what is the correct behavior here, especially in the case of heterogeneous collections. Checking just the first type could produce some really od behaviors with `Sort-Object -Unique` where the order of the object changes. I will close the PR for now until we agree to follow this direction.